Sources/Compiler/TypedNode.swift

```diff
@@ -138,9 +143,57 @@ extension TypedNode where ID: ExprID {
 }
 
 extension TypedNode where ID == NodeID<NameExpr> {
+  enum Domain: Equatable {
+
+    /// No domain.
+    case none
+
+    /// Domain is implicit; the expression denotes a type member.
+    case implicit
+
+    /// Domain is a value expression or a type identifier.
+    case expr(TypedNode<AnyExprID>)
+
+  }
+
+  /// The domain of the name, if it is qualified.
```

Comment:
I realize this is a copy/paste, but is it accurate, @kyouko-taiga ? What does `none` mean if not unqualified?  This property isn't an optional!

Reply:
You can think of this type as extending an optional with a third case. In other words, I'm flattening `Optional<Implicit | AnyExprID>`. Perhaps we could make `Name.Domain` conform to `ExpressibleByNilLiteral` to further underline the role of this type as some form of optional.

Reply:
Then the comment is wrong.  It should be
```suggestion
  /// The name's qualification.
```

There is no "if."

Reply:
Could you have a look at https://github.com/val-lang/val/pull/147?

I think the "if" is consistent with the way other "optional" properties have been documented. As mentioned, `NameExpr.Domain` is a kind of optional type, only with two non-nil cases.

If you still think there shouldn't be an "if", then could you suggest this edit in the other PR?